### DISCUSSION TITLE
Made an error in reverting my git commit

### DISCUSSION THREAD

**Discussion by dmastrop** [community/community#62726] [Programming Help] [answered]
### Body

hi All

Hoping someone can help get me back to state where I want to get on this....

I accidentally reverted a commit that I did not want to revert using git revert <commit-id>

Actually I just wanted to get rid of my uncommitted and unstaged work and go back to the last commit, not revert the last commit. I tried using a 
git add . (adding everything)
git stash 
git stash drop

to get rid of all the latest uncommitted changes and to restore state to last commit but something went wrong and it did not look right.

I have not made any commits since this last commit that I want to restore to that commit.

How can I get back to this commit state.

I know it is a simple solution but I don't want to make the situation worse ;)

Will a simple 
revert a05f341df1578822238d56d2777fdb230a2d8bba  do it?

I don't want all the uncommitted work that was on the workspace at the time that I did that. I just want to get back to the 7e73 commit below with the grafana-destroy.yml file present

It is missing right now.

Also if I have unstaged and/or uncommitted stuff and I just want to get rid of it, how do I get rid of it (easiest way). I thought the git stash approach would have worked but it did not.

thanks and warm regards
Dave

=========

**Here is the git log from what happened.**.... I want to restore to the ed7e73 commit below. right now it is gone from my workspace

ave:~/environment/mtc-terraform-ansible-jenkins (master) $ **git revert ceb344ada2d6ef96432036e958207f2232ed7e73** 
[master a05f341] Revert "added a grafana-destroy.yml file with relevant changes in the yaml syntax to do the effective destroy of the granfana.yml. NOTE: must first stop grafana-service BEFORE you can remove the installation. It will NOT get removed if the service is not explicitly stoppedgit add ."
 Committer: Ubuntu <ubuntu@ip-172-31-26-107.us-west-1.compute.internal>
Your name and email address were configured automatically based
on your username and hostname. Please check that they are accurate.
You can suppress this message by setting them explicitly:

    git config --global user.name "Your Name"
    git config --global user.email you@example.com

After doing this, you may fix the identity used for this commit with:

    git commit --amend --reset-author

 2 files changed, 1 insertion(+), 91 deletions(-)
 delete mode 100644 playbooks/grafana-destroy.yml
dave:~/environment/mtc-terraform-ansible-jenkins (master) $ git log
commit a05f341df1578822238d56d2777fdb230a2d8bba (HEAD -> master)
Author: Ubuntu <ubuntu@ip-172-31-26-107.us-west-1.compute.internal>
Date:   Sat Aug 5 03:08:09 2023 +0000

    Revert "added a grafana-destroy.yml file with relevant changes in the yaml syntax to do the effective destroy of the granfana.yml. NOTE: must first stop grafana-service BEFORE you can remove the installation. It will NOT get removed if the service is not explicitly commit a05f341df1578822238d56d2777fdb230a2d8bba (HEAD -> master)
Author: Ubuntu <ubuntu@ip-172-31-26-107.us-west-1.compute.internal>
Date:   Sat Aug 5 03:08:09 2023 +0000
commit a05f341df1578822238d56d2777fdb230a2d8bba (HEAD -> master)
Author: Ubuntu <ubuntu@ip-172-31-26-107.us-west-1.compute.internal>
Date:   Sat Aug 5 03:08:09 2023 +0000

    Revert "added a grafana-destroy.yml file with relevant changes in the yaml syntax to do the effective destroy of the granfana.yml. 
NOTE: must first stop grafana-service BEFORE you can remove the installation. It will NOT get removed if the service is not explicitly 
stoppedgit add ."
    
    This reverts commit ceb344ada2d6ef96432036e958207f2232ed7e73.
commit a05f341df1578822238d56d2777fdb230a2d8bba (HEAD -> master)
Author: Ubuntu <ubuntu@ip-172-31-26-107.us-west-1.compute.internal>
Date:   Sat Aug 5 03:08:09 2023 +0000

    Revert "added a grafana-destroy.yml file with relevant changes in the yaml syntax to do the effective destroy of the granfana.yml. 
NOTE: must first stop grafana-service BEFORE you can remove the installation. It will NOT get removed if the service is not explicitly 
stoppedgit add ."
    
    This reverts commit ceb344ada2d6ef96432036e958207f2232ed7e73.

commit ceb344ada2d6ef96432036e958207f2232ed7e73 (origin/master)
Author: Ubuntu <ubuntu@ip-172-31-26-107.us-west-1.compute.internal>
Date:   Sat Aug 5 02:10:58 2023 +0000

=======

**here is my current git log status**

commit a05f341df1578822238d56d2777fdb230a2d8bba (HEAD -> master)  **<<<< I want to revert this to my last commit below, and I don't want whatever uncommitted work I had in the workspace at the time I did this.**

Author: Ubuntu <ubuntu@ip-172-31-26-107.us-west-1.compute.internal>
Date:   Sat Aug 5 03:08:09 2023 +0000

    Revert "added a grafana-destroy.yml file with relevant changes in the yaml syntax to do the effective destroy of the granfana.yml. 
NOTE: must first stop grafana-service BEFORE you can remove the installation. It will NOT get removed if the service is not explicitly 
stoppedgit add ."
    
    This reverts commit ceb344ada2d6ef96432036e958207f2232ed7e73.

commit ceb344ada2d6ef96432036e958207f2232ed7e73 (origin/master)
Author: Ubuntu <ubuntu@ip-172-31-26-107.us-west-1.compute.internal>
Date:   Sat Aug 5 02:10:58 2023 +0000

    added a grafana-destroy.yml file with relevant changes in the yaml syntax to do the effective destroy of the granfana.yml. NOTE: mu
st first stop grafana-service BEFORE you can remove the installation. It will NOT get removed if the service is not explicitly stoppedg
it add .

commit fe9b37691f38205ed58cc27990191080ff2e136e

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by dmastrop** [comment]
Would a 

git reset --hard ceb344ada2d6ef96432036e958207f2232ed7e73 

get me back to the 7e73 state without all the unstaged and uncommitted work on my workspace?

**Comment by Spixmacaw** [comment]
Yes you will have to resolve the conflicts, mark them as so with git add or git rm and git commit
after the git revert - if you see .git/MERGE_MSG, you will see something like:

Revert "adding revert"

This reverts commit c1366607f15a8384434948cb0bcbf8ece48bb460.

Conflicts:

revert

**Reply by dmastrop** [reply]
hi @Spixmacaw 

I think your reply was for another thread.

warm regards

Dave

**Reply by Spixmacaw** [reply]
i ans for given ques

**Reply by dmastrop** [reply]
I don't understand your answer. The commit id that you listed is not one of my commits.

There are no conflicts. I just need to get back to the commit ceb344ada2d6ef96432036e958207f2232ed7e73 (and without the changes that I had on my workspace at the time of the commit; I don't need those changes and want them discarded)

Would a 
git reset --hard ceb344ada2d6ef96432036e958207f2232ed7e73  do this?

warm regards

Dave

**Reply by dmastrop** [reply]
I created a branch and did the git reset --hard ceb344ada2d6ef96432036e958207f2232ed7e73 and it did what I needed it to do.

(I don't care that the history of the bad commit (the revert) was removed. I know that the git reset --hard will do this.)

warm regards

Dave

**Comment by Mariyajoseph24** [comment, marked as answer]
Hello @dmastrop , hope this helps
First, to get rid of all the unstaged and uncommitted changes, you can use the following command:
```
git reset --hard HEAD
```
This will reset your working directory to the last commit, discarding all the changes you made since the last commit.

Now, to restore your state to a specific commit (7e73 in your case), you can use the following command:
```
git reset --hard 7e73
```
This will reset your working directory and move the HEAD to the specified commit, effectively restoring your state to that commit.

As for the revert command you mentioned:
```
git revert a05f341df1578822238d56d2777fdb230a2d8bba
```
This command will create a new commit that undoes the changes introduced by the specified commit (a05f341df1578822238d56d2777fdb230a2d8bba). However, since you want to get back to the state of the commit below (7e73) without creating any new commits, you don't need to use the revert command.

Remember, using `git reset --hard` and `git revert` can be powerful commands, so be cautious when using them. Always make sure you have a backup or have committed any important changes before running these commands.

If you ever need to get rid of unstaged and uncommitted stuff, the `git reset --hard HEAD` command should do the trick.

I hope this helps!

**Reply by dmastrop** [reply]
Thanks Mariya  for the clear and complete and concise answer. Much appreciated.
I was in a rush and ended up just doing the 

git reset --hard 7e73

This worked great. I think all of the other uncommitted and unstaged stuff was already gone because I did a failed stash procedure that got me into the problem.

But now I know to get rid of that uncommitted and unstaged stuff first do the 
git reset --hard HEAD

then reset to the commit that I want.

As you said this is very powerful and dangerous.  I did a branch before I did the reset in case something happened.

thanks again

warmest regards....
Dave
